File: EventFlowAPI/EventFlowAPI.Logic/DTO/Statistics/ResponseDto/HallRentStatistics.cs
using EventFlowAPI.DB.Entities;

namespace EventFlowAPI.Logic.DTO.Statistics.ResponseDto
{
    public class HallRentStatistics
    {
        public int AllAddedHallRentsCount { get; set; }
        public int AllHallRentsThatTookPlaceInTimePeriod { get; set; }
        public int AllCanceledHallRentsCount { get; set; }
        public TimeSpan DurationAvg { get; set; }
        public double TotalHallRentsIncome { get; set; }
        public Dictionary<string, double> UserReservationsDict { get; set; } = [];
        public Dictionary<string, double> HallReservationsDict { get; set; } = [];
        public Dictionary<string, double> HallTypeReservationsDict { get; set; } = [];
        public Dictionary<string, double> HallAddtionalServicesReservationsDict { get; set; } = [];
    }
}

File: EventFlowClient/src/models/response_models/stats/HallRentStatistics.ts
export type HallRentStatistics = {
  allAddedHallRentsCount: number;
  allHallRentsThatTookPlaceInTimePeriod: number;
  allCanceledHallRentsCount: number;
  durationAvg: number;
  totalHallRentsIncome: number;
  userReservationsDict: Record<string, number>;
  hallReservationsDict: Record<string, number>;
  hallTypeReservationsDict: Record<string, number>;
  hallAddtionalServicesReservationsDict: Record<string, number>;
};
